### ISSUE TITLE
no-invalid-regexp should syntax-check regexp literals

### ISSUE THREAD

**Issue by clar-cmp** [eslint/eslint#12841]
**What rule do you want to change?** `no-invalid-regexp`

**Does this change cause the rule to produce more or fewer warnings?** More warnings

**How will the change be implemented? (New option, new default behavior, etc.)?** The change will be modified to include both literals and strings passed to the constructor

**Please provide some example code that this change will affect:**

```js
const regexp = /(/;
```

**What does the rule currently do for this code?** Absolutely nothing

**What will the rule do after it's changed?** It should trigger the rule, noting that the unmatched `(` is invalid syntax.

**Are you willing to submit a pull request to implement this change?** Don't have time for it; up for grabs

**Comment by kaicataldo** [comment]
From the [rule docs](https://eslint.org/docs/rules/no-invalid-regexp):

> An invalid pattern in a regular expression literal is a SyntaxError when the code is parsed, but an invalid string in RegExp constructors throws a SyntaxError only when the code is executed.

This looks like it's caught by the default parser (example [here](https://eslint.org/demo#eyJ0ZXh0IjoiY29uc3QgcmVnZXhwID0gLygvOyIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6MTEsInNvdXJjZVR5cGUiOiJtb2R1bGUiLCJlY21hRmVhdHVyZXMiOnsianN4Ijp0cnVlfX0sInJ1bGVzIjp7fSwiZW52Ijp7ImJyb3dzZXIiOmZhbHNlfX19)). If you're using a custom parser, this would be a great issue to file with the maintainers of that project!

**Comment by clar-cmp** [comment]
Oh, I completely missed this. I'll fill a bug with `typescript-eslint`.
